src/Features/LanguageServer/Protocol/LspServices/LspServices.cs

```diff
@@ -28,37 +28,69 @@ internal class LspServices : IDisposable
         ImmutableArray<Lazy<ILspService, LspServiceMetadataView>> mefLspServices,
         ImmutableArray<Lazy<ILspServiceFactory, LspServiceMetadataView>> mefLspServiceFactories,
         WellKnownLspServerKinds serverKind,
-        ImmutableArray<Lazy<ILspService, LspServiceMetadataView>> baseServices)
+        ImmutableArray<Lazy<ILspService, LspServiceMetadataView>> baseServices,
+        IServiceCollection serviceCollection)
     {
         // Convert MEF exported service factories to the lazy LSP services that they create.
         var servicesFromFactories = mefLspServiceFactories.Select(lz => new Lazy<ILspService, LspServiceMetadataView>(() => lz.Value.CreateILspService(this, serverKind), lz.Metadata));
 
         var services = mefLspServices.Concat(servicesFromFactories);
 
+#if DEBUG
+        var duplicateServices = services.GroupBy(s => s.Metadata.Type).Where(s => s.Count() > 1);
```

Comment:
Any reason this can't just be a test? Our tests generally run on the entire MEF composition, so should hopefully be possible.

Reply:
> Our tests generally run on the entire MEF composition

Could you point me to some of those tests?

Reply:
Ha, as soon as I send that message I realize where they are, nevermind.

Reply:
For future reference, I removed this check because we'll get an exception from the `ToImmutableDictionary` call anyway.